classes/record/RecordDAO.inc.php

```diff
@@ -264,13 +264,27 @@ function getInsertRecordId() {
 	/**
 	 * Retrieve a count of the records available.
 	 * @param $archiveId Specific archive to query (optional)
+	 * @param $onlyEnabled boolean
 	 * @return int
 	 */
-	function getRecordCount($archiveId = null) {
-		$result =& $this->retrieve(
-			'SELECT COUNT(*) AS count FROM records' . (isset($archiveId)?' WHERE archive_id = ?':''),
-			isset($archiveId)?$archiveId:false
-		);
+	function getRecordCount($archiveId = null, $onlyEnabled = false) {
+		$sql = 'SELECT COUNT(*) AS count
+				FROM records r';
+
+		if ($onlyEnabled) $sql .= ' LEFT JOIN archives a ON r.archive_id = a.archive_id WHERE a.enabled = 1';
```

Comment:
This is a little convoluted -- can you rewrite this so we don't need the `$archiveId` to be implemented differently for each `$onlyEnabled` case?